src/Momento.Sdk.Incubating/Responses/CacheDictionaryGetBatchResponse.cs

```diff
@@ -1,40 +1,101 @@
 ﻿using System.Collections.Generic;
 using System.Linq;
+using Google.Protobuf.Collections;
 using Momento.Protos.CacheClient;
+using Momento.Sdk.Exceptions;
 using Momento.Sdk.Responses;
-
+using static Momento.Protos.CacheClient._DictionaryGetResponse.Types;
 
 namespace Momento.Sdk.Incubating.Responses;
 
-public class CacheDictionaryGetBatchResponse
+public abstract class CacheDictionaryGetBatchResponse
 {
-    public IEnumerable<CacheDictionaryGetResponse> Responses { get; private set; }
-
-    public CacheDictionaryGetBatchResponse(_DictionaryGetResponse response, int numRequested)
+    public class Success : CacheDictionaryGetBatchResponse
     {
-        if (response.DictionaryCase == _DictionaryGetResponse.DictionaryOneofCase.Found)
+        public List<CacheDictionaryGetResponse> Responses { get; private set; }
+
+        public Success(_DictionaryGetResponse responses)
         {
-            Responses = response.Found.Items.Select(item => new CacheDictionaryGetResponse(item));
+            var responsesList = new List<CacheDictionaryGetResponse>();
+            foreach (_DictionaryGetResponsePart response in responses.Found.Items)
+            {
+                if (response.Result == ECacheResult.Hit)
+                {
+                    responsesList.Add(new CacheDictionaryGetResponse.Hit(response.CacheBody));
+                }
+                if (response.Result == ECacheResult.Miss)
+                {
+                    responsesList.Add(new CacheDictionaryGetResponse.Miss());
+                }
```

Comment:
This should be an error if neither of these is true. Shouldn't happen but there are other values in the `ECacheResult` enumerable